pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/ExtractTransformFunction.java

```diff
@@ -0,0 +1,91 @@
+package org.apache.pinot.core.operator.transform.function;
+
+import java.util.List;
+import java.util.Map;
+import org.apache.pinot.core.operator.blocks.ProjectionBlock;
+import org.apache.pinot.core.operator.transform.TransformResultMetadata;
+import org.apache.pinot.segment.spi.datasource.DataSource;
+import org.apache.pinot.spi.data.FieldSpec;
+import org.joda.time.Chronology;
+import org.joda.time.DateTimeField;
+import org.joda.time.chrono.ISOChronology;
+
+
+public class ExtractTransformFunction extends BaseTransformFunction {
+  public static final String _name = "extract";
+  private TransformResultMetadata _resultMetadata;
+  private TransformFunction _mainTranformFunction;
+  private String[] _stringOutputTimes;
+  protected String _field;
+  protected Chronology _chronology = ISOChronology.getInstanceUTC();
+  
+  private static final TransformResultMetadata METADATA =
+      new TransformResultMetadata(FieldSpec.DataType.INT, true, false);
+  
+  @Override
+  public String getName() { return _name; }
+
+  @Override
+  public void init(List<TransformFunction> arguments, Map<String, DataSource> dataSourceMap) {
+    if (arguments.size() != 2) { 
+      throw new IllegalArgumentException("Exactly 2 arguments are required for EXTRACT transform function"); 
+    }
+
+    _field = ((LiteralTransformFunction) arguments.get(0)).getLiteral();
+    _mainTranformFunction = arguments.get(1);
+  }
+
+  @Override
+  public TransformResultMetadata getResultMetadata() {
+    return _resultMetadata;
```

Comment:
We already have a constant for it: `INT_SV_NO_DICTIONARY_METADATA`
```suggestion
    return INT_SV_NO_DICTIONARY_METADATA;
```

Reply:
Thanks for the suggestion!